File: C/nth_fibonacci_number.c
#include <stdio.h>

int main() {
    int n;
    long long a = 0, b = 1, c;

    printf("Enter the value of n: ");
    scanf("%d", &n);

    if (n <= 0) {
        printf("Invalid input\n");
    } 
    else if (n == 1) {
        printf("The %dth Fibonacci number is %lld\n", n, a);
    } 
    else if (n == 2) {
        printf("The %dth Fibonacci number is %lld\n", n, b);
    } 
    else {
        for (int i = 3; i <= n; i++) {
            c = a + b;
            a = b;
            b = c;
        }
        printf("The %dth Fibonacci number is %lld\n", n, b);
    }

    return 0;
}

File: Python/nth_fibonacci_number.py
def main():
    n = int(input("Enter the value of n: "))

    a = 0
    b = 1

    if n <= 0:
        print("Invalid input")
    elif n == 1:
        print(f"The {n}th Fibonacci number is {a}")
    elif n == 2:
        print(f"The {n}th Fibonacci number is {b}")
    else:
        for _ in range(3, n + 1):
            c = a + b
            a = b
            b = c
        print(f"The {n}th Fibonacci number is {b}")


if __name__ == "__main__":
    main()

